### DISCUSSION TITLE
How to auto-fix typo using buffer words

### DISCUSSION THREAD

**Discussion by nyngwang** [hrsh7th/nvim-cmp#998] [Q&A] [answered]
For example I usually type `tf.keras` into `tf.kersa`, but type `tf.` and wait for the completion list to show `keras` showing as an item and choose it is just too slow. Is it possible to `tf.kersa` auto-fix if nvim-cmp somehow could know that `kersa` is a typo of `keras` and auto-fix/auto-choose it for me?

Maybe this feature has been implemented via another plugin, I'm not sure.

**Comment by Shougo** [comment, marked as answer]
It seems hard.  Because nvim-cmp does not understand what is the correct word.
Use LSP is better for it.  LSP knows syntax.

**Reply by nyngwang** [reply]
I have tried some options before reading your answer. It's really hard in a sense that some complicated ranking algorithms are involved, e.g. editing-distance. (even this one is just my guessing)

**Reply by nyngwang** [reply]
I just realized that: Any feature requiring guessing users' intent is very hard. Sorry for my bad question.

**Reply by nyngwang** [reply]
FYI: I just summoned the demon github copilot that seems to remedy this requirement.
